kafka/producer.go

```diff
@@ -204,6 +204,7 @@ func (p *Producer) Produce(ctx context.Context, rs ...apmqueue.Record) error {
 		kgoRecord := &kgo.Record{
 			Headers: headers,
 			Topic:   string(record.Topic),
+			Key:     record.OrderingKey,
```

Comment:
It's not a comment directly related to this change, but why are we having this option? AFAIK we haven't used the key to route messages to a partition to avoid hot-spotting. I'm curious to know what use cases we'd need this(stateful processing?)

Reply:
We have a use case where we want a set of messages to be consumed by the same instance of the consumer.